renku/core/management/datasets.py

```diff
@@ -110,25 +110,36 @@ def load_dataset(self, name=None):
     def with_dataset(self, name=None):
         """Yield an editable metadata object for a dataset."""
         dataset = self.load_dataset(name=name)
+        clean_up_required = False
 
         if dataset is None:
+            clean_up_required = True
+            dataset_ref = None
             identifier = str(uuid.uuid4())
             path = (self.renku_datasets_path / identifier / self.METADATA)
-            path.parent.mkdir(parents=True, exist_ok=True)
+            path.parent.mkdir(parents=True, exist_ok=False)
```

Comment:
This will produce a non-click exception. Do we have a handler for it?

Reply:
Will be addressed in https://github.com/SwissDataScienceCenter/renku-python/issues/720.